src/services/formatting/smartIndenter.ts

```diff
@@ -429,6 +429,7 @@ namespace ts.formatting {
                 case SyntaxKind.ArrayBindingPattern:
                 case SyntaxKind.ObjectBindingPattern:
                 case SyntaxKind.JsxElement:
+                case SyntaxKind.FunctionType:

```

Comment:
we should also add:
- ConstructorType
- ConstructSignature
- CallSignature
- ParenthesizedType
- TaggedTemplateExpression
- AwaitExpression
- YieldExpression